DokuwikiXMLExport.php

```diff
@@ -0,0 +1,194 @@
+<?php
+
+/**
+ *  This is the Dokuwiki export made by D. Brader for FINDOLOGIC.
+ *  If any bugs occur, please contact the support team (support@findologic.com).
+ */
+if(!defined('DOKU_INC')) {
+    define('DOKU_INC',realpath(dirname(__FILE__).'/../../../').'/');
+}
+
+require_once (DOKU_INC.'inc/init.php');
+require (__DIR__ . '/vendor/autoload.php');
+
+use FINDOLOGIC\Export\Exporter;
+use FINDOLOGIC\Export\Data\Name;
+use FINDOLOGIC\Export\Data\Summary;
+use FINDOLOGIC\Export\Data\Description;
+use FINDOLOGIC\Export\Data\Price;
+use FINDOLOGIC\Export\Data\Url;
+use FINDOLOGIC\Export\Data\Ordernumber;
+use FINDOLOGIC\Export\Data\DateAdded;
+use FINDOLOGIC\Export\Data\Attribute;
+
+class DokuwikiXMLExport
+{
+    /**
+     * Generate the entire XML Export based on the Dokuwiki metadata.
+     *
+     * @param $start integer Determines the first item (offset) to be exported.
+     * @param $count integer Determines the interval size / number of items to be exported.
+     * @return string Returns the XML.
+     */
+    public function generateXMLExport($start, $count)
+    {
+        global $conf;
+
+        $Indexer = new Doku_Indexer();
+        $pagesAndDeletedPages = $Indexer->getPages();
+
+        $excludedPages = $this->splitConfigToArray($conf['plugin']['findologicxmlexport']['excludePages']);
+        $pages = null;
```

Comment:
Maybe make `$pages` an instance variable and fill it in the constructor. Then you won't have to pass `$pages` to every method eg. `getName`.

Reply:
TODO:

- Define variable in constructor and make it available. After one page is finished the Class knows, so I can define something that says "hey I am done here" and deletes the page from the array or something like this.

Reply:
@georgms I have made an issue for this. See #10.